### PR TITLE
Function Calling Support for Gemini - Part 2

### PR DESCRIPTION
**Pull request by luxzoli**
## Why are these changes needed?

Finalize the Gemini function calling PR #2793 from @arjun-g, plus include support for VertexAI tool config for function calling. 

## Related issue number

Relates to #2387

## Checks

- [ ] I've included any doc changes needed for https://microsoft.github.io/autogen/. See https://microsoft.github.io/autogen/docs/Contribute#documentation to build and test documentation locally.
- [x] I've added tests (if relevant) corresponding to the changes introduced in this PR.
- [x] I've made sure all auto checks have passed.

### CODE REVIEW

**Review by ekzhu — APPROVED** [review]
Looks fine with me though we cannot test Gemini in our CI.

File: autogen/oai/gemini.py [hunk 1]
```diff
@@ -340,34 +418,32 @@ def _oai_content_to_gemini_content(self, content: Union[str, List]) -> List:
...
-                concatenated_parts.append(previous_part)
-                previous_part = current_part
-
-        if previous_part.text == "":
-            if self.use_vertexai:
-                previous_part = VertexAIPart.from_text("empty")
-            else:
-                previous_part.text = "empty"  # Empty content is not allowed.
-        concatenated_parts.append(previous_part)
+    def _calculate_gemini_cost(self, input_tokens: int, output_tokens: int, model_name: str) -> float:
```
Comment (ekzhu) [inline autogen/oai/gemini.py:421]: This calculation needs to be maintained @luxzoli

**Comment by ekzhu** [comment]
@luxzoli Thanks! A couple of tests are failing in CI.

**Comment by luxzoli** [comment]
@ekzhu should I prepare a new PR with the [updated costs](https://cloud.google.com/vertex-ai/generative-ai/pricing)? They are indeed slightly different now for Gemini 1.5 Pro and Gemini 1.5 Flash

**Pull request merged**